wordpress-mobile/WordPress-iOS
# Refactor `SharingService` to use `CoreDataStack`

Use `CoreDataStack` instead of `NSManagedObjectContext` in `SharingService`. See the "Issue" and "Proposal" sections of https://github.com/wordpress-mobile/WordPress-iOS/pull/19893 for rationale behind this change.

Like other similar PRs, I recommend reviewing this one commit by commit too.

## Test Instructions

Only the "Sharing" screen (under the My Site tab) is affected by this change. Verifying the features in it still works should be enough, i.e. connect/disconnect to a SNS, show/hide sharing buttons.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What are describe in the "Test Instructions" section above.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:
